app/questionnaire/navigator.py

```diff
@@ -39,8 +39,9 @@ def is_interstitial_block(cls, block_id):
     @classmethod
     def _block_index_for_location(cls, blocks, location):
         if not cls.is_interstitial_block(location['block_id']):
-            return next(index for (index, b) in enumerate(blocks) if b["block"]["id"] == location['block_id'] and
-                        b["group_id"] == location['group_id'] and b['group_instance'] == location['group_instance'])
+            return next(index for (index, b) in enumerate(blocks) if
```

Comment:
This seems dangerous to me. A lot of the repeat logic is based around being within the same group/instance. It's not clear what impact removing the group id check has. Seems like routing rules should be able to specify both group/block_id.